mlflow/deployments/databricks/__init__.py

```diff
@@ -207,6 +247,47 @@ def predict(self, deployment_name=None, inputs=None, endpoint=None):
             timeout=MLFLOW_DEPLOYMENT_PREDICT_TIMEOUT.get(),
         )
 
+    @experimental
+    def predict_stream(self, deployment_name=None, inputs=None, endpoint=None) -> Iterator[Dict[str, Any]]:
+        """
+        Submit a query to a configured provider endpoint, and get streaming response
+
+        Args:
+            deployment_name: Unused.
+            inputs: The inputs to the query, as a dictionary.
+            endpoint: The name of the endpoint to query.
+
+        Returns:
+            An iterator of dictionary containing the response from the endpoint.
+        """
+        inputs = inputs or {}
+
+        # Add stream=True param in request body to get streaming response
+        # See https://docs.databricks.com/api/workspace/servingendpoints/query#stream
+        chunk_line_iter = self._call_endpoint_stream(
+            method="POST",
+            prefix="/",
+            route=posixpath.join(endpoint, "invocations"),
+            json_body={**inputs, "stream": True},
+            timeout=MLFLOW_DEPLOYMENT_PREDICT_TIMEOUT.get(),
+        )
+
+        def result_gen_fn():
+            for line in chunk_line_iter:
+                key, value = line.split(":", 1)
+                if key != "data":
+                    raise MlflowException("Unknown response format.")
+
+                value = value.strip()
+                if value == "[DONE]":
+                    # Databricks endpoint streaming response ends with
+                    # a line of "data: [DONE]"
+                    return
+
+                yield json.loads(value)
+
+        return result_gen_fn()
```

Comment:
Do we need `result_gen_fn`?

Reply:
We need to throw away the last line "data: [DONE]", generator should be a easy way to do it.

Reply:
What would happen if we remove `result_gen_fn`?

Reply:
Oh . Got it. you are right, we can remove it.